### DISCUSSION TITLE
State of alerts with same name

### DISCUSSION THREAD

**Discussion by dishonono** [prometheus/prometheus#11470] [Q&A]
Hi,

Question about different alert rules with same name: It is possible in practice to define 2 or more alerts with same name. They can be defined in different rule groups.

Lets look at a case where these rules have different query and annotations, but outputs same label set. 
The different annotation may include different notification message. Everything will work as expected for this user at first. He will see the 2 alerts outputing notifcations as expected.
But since the label set is the same, then both ALERTS and ALERTS_FOR_STATE metric of these 2 rules will be mixed. As long as the promeheus server is running, they will have sepeare state, but if the server is restarted, the state will be restored from ALERTS_FOR_STATE , and both rules will get the same state. Then the user will start experiencing strange behiviours like delayed alerts.

Is this a known issue? Should customers avoid using same alert name unless they absolutly know what they are doing? Why not at  include the rule group name and the rule index in the group in the the alertForState labels?

**Comment by Shreyanth07** [comment]
Yes you can do that as far as I have tested
